#!/usr/bin/env python
PACKAGE = "delta_manipulator"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("v_traj", double_t, 0, "trajectory_speed",    0.1, 0.0,   1.0)
gen.add("r_traj", double_t, 0, "trajectory_radius",    0.5, 0.01,   1.0)
gen.add("h_traj", double_t, 0, "trajectory_altitude",    0.5, 0.0,   3.0)
gen.add("x_traj", double_t, 0, "trajectory_x_origin",    0.0, -3.0,   3.0)
gen.add("y_traj", double_t, 0, "trajectory_y_origin",    0.0, -3.0,   3.0)
gen.add("yaw_traj", double_t, 0, "trajectory_initial_yaw_angle",    0.0, -3.2,   3.2)

gen.add("repeats", int_t, 0, "number of times to perform a trajectory",  2 , 1,   10)

mode_enum = gen.enum([gen.const("static", int_t, 0, "static position hold"),
                 gen.const("line_z", int_t, 1, "line trajectory in z"),
                 gen.const("line_y", int_t, 2, "line trajectory in y"),
                 gen.const("line_x", int_t, 3, "line trajectory in x"),
                 gen.const("circle_xy", int_t, 4, "circular trajectory in xy plane"),
                 gen.const("circle_xz", int_t, 5, "circular trajectory in xz plane"),
                 gen.const("figure_8", int_t, 6, "figure 8 trajectory")],
                 "select trajectory type")

gen.add("drone_mode", int_t, 0, "mode selector", 0, 0, 10, edit_method=mode_enum)

gen.add("v_traj_delta", double_t, 0, "delta_trajectory_speed",    0.25, 0.0,   1.0)
gen.add("r_traj_delta", double_t, 0, "delta_trajectory_radius",    0.12, 0.01,   0.5)
gen.add("h_traj_delta", double_t, 0, "delta_trajectory_height",    0.5, 0.0,   3.0)

gen.add("delta_mode", int_t, 0, "mode selector", 0, 0, 10, edit_method=mode_enum)

exit(gen.generate(PACKAGE, "delta_manipulator", "Trajectory"))